docs/src/main/asciidoc/spring-cloud-stream.adoc

```diff
@@ -709,7 +709,60 @@ public Function<List<Person>, Person> findFirstPerson() {
 }
 ----
 
-==== Annotation-based support
+===== Spring Integration flow as functions
+There are times when implementing a function you may have complex requirements that fit the category of
+https://www.enterpriseintegrationpatterns.com[Enterprise Integration Patterns] (EIP) and as such
+best to be handled using framework such as https://spring.io/projects/spring-integration[Spring Integration] (SI) -
+a reference implementation of EIP.
+
+Thankfully SI already provides support for exposing integration flows as functions.
+Let's look at the sample:
+
+[source, java]
+----
+@SpringBootApplication
+public class FunctionSampleSpringIntegrationApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(FunctionSampleSpringIntegrationApplication.class, args);
+	}
+
+	@Bean
+	public IntegrationFlow uppercaseFlow() {
+		return IntegrationFlows.from(MessageFunction.class, "uppercase")
+				.<String, String>transform(String::toUpperCase)
+				.logAndReply(LoggingHandler.Level.WARN);
+	}
+
+	public interface MessageFunction extends Function<Message<String>, Message<String>> {
+
+	}
+}
+----
+
+For those who are familiar with SI you can see we are simply defining a bean of type `IntegrationFlow` where we
+declaring an integration flow we want to expose as a `Function<String, String>` (using SI DSL) by the name `uppercase`.
```

Comment:
Change to "declare an integration flow that we want to expose as a `Function<String, String>` (using SI DSL) called"